frontend/amundsen_application/static/js/features/ColumnList/ColumnType/parser.ts

```diff
@@ -191,32 +208,29 @@ export function convertNestedTypeToColumns(
     if (typeof child === 'string') {
       const [columnName, colType] = child.split(COLUMN_TYPE_SEPARATOR);
       if (colType !== undefined) {
-        nestedColumns.push({
-          badges: [],
-          col_type: colType.replace(SEPARATOR_DELIMETER, ''),
-          description: '',
-          name: columnName,
-          sort_order: 0,
-          nested_level: nestedLevel,
-          is_editable: false,
-          stats: [],
-        });
+        const column = createNestedColumn(
+          columnName,
+          colType.replace(SEPARATOR_DELIMETER, ''),
+          nestedLevel
+        );
+        nestedColumns.push(column);
       }
     } else {
+      // Changing nestedLevel introduces a scope-level bug.
+      let incNestedLevel = 0;
       if (child.name !== '') {
-        nestedColumns.push({
-          badges: [],
-          col_type: child.col_type || '',
-          description: '',
-          name: child.name || '',
-          sort_order: 0,
-          nested_level: nestedLevel,
-          is_editable: false,
-          stats: [],
-        });
-        nestedLevel++;
```

Comment:
This ++ here increments future references to the same variable since the scope is apparently shared.

Reply:
Probably wouldn't be an issue for `()=>` syntax functions vs `function` syntax